bitnami/superset/templates/flower/deployment.yaml

```diff
@@ -0,0 +1,189 @@
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.flower.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
+      labels: {{- include "common.labels.standard" ( dict "customLabels" $podLabels "context" $ ) | nindent 8 }}
+        app.kubernetes.io/component: celery-flower
+        app.kubernetes.io/part-of: superset
+    spec:
+      {{- include "superset.imagePullSecrets" . | nindent 6 }}
+      serviceAccountName: {{ template "superset.serviceAccountName" . }}
+      automountServiceAccountToken: {{ .Values.flower.automountServiceAccountToken }}
+      {{- if .Values.flower.hostAliases }}
+      hostAliases: {{- include "common.tplvalues.render" (dict "value" .Values.flower.hostAliases "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.affinity }}
+      affinity: {{- include "common.tplvalues.render" ( dict "value" .Values.flower.affinity "context" $) | nindent 8 }}
+      {{- else }}
+      affinity:
+        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.flower.podAffinityPreset "component" "celery-flower" "customLabels" $podLabels "context" $) | nindent 10 }}
+        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.flower.podAntiAffinityPreset "component" "celery-flower" "customLabels" $podLabels "context" $) | nindent 10 }}
+        nodeAffinity: {{- include "common.affinities.nodes" (dict "type" .Values.flower.nodeAffinityPreset.type "key" .Values.flower.nodeAffinityPreset.key "values" .Values.flower.nodeAffinityPreset.values) | nindent 10 }}
+      {{- end }}
+      {{- if .Values.flower.nodeSelector }}
+      nodeSelector: {{- include "common.tplvalues.render" ( dict "value" .Values.flower.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.tolerations }}
+      tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.flower.tolerations "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.priorityClassName }}
+      priorityClassName: {{ .Values.flower.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.flower.schedulerName }}
+      schedulerName: {{ .Values.flower.schedulerName | quote }}
+      {{- end }}
+      {{- if .Values.flower.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.flower.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.podSecurityContext.enabled }}
+      securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.flower.podSecurityContext "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.flower.terminationGracePeriodSeconds }}
+      {{- end }}
+      initContainers:
+        {{- if .Values.waitForDB.enabled }}
```

Comment:
In newer charts we started creating a scetion called defaultInitContainers where we group all these init containers. You can check dremio for example.